File: DB/db_backup_restore/python_bak_restore/restore.py
import subprocess
import shutil
import os
import logging
from DB.python_backup_restore.python_bak_restore.config import backup_dir

# 设置日志记录
logging.basicConfig(filename='backup_restore.log', level=logging.ERROR)

def restore_mysql(host, username, password, database, backup_file):
    try:
        # 使用安全的参数传递方式
        subprocess.run(['mysql', '-h', host, '-u', username, '-p' + password, database, '<', backup_file], check=True, shell=True)
    except subprocess.CalledProcessError as e:
        logging.error(f"Error occurred during MySQL restore: {e}")

def restore_oracle(username, password, sid, backup_file):
    try:
        shutil.copy(backup_file, os.path.join(backup_dir, f'{sid}_backup.dmp'))
        # 使用安全的参数传递方式
        subprocess.run(['imp', username + '/' + password + '@' + sid, 'file=' + os.path.join(backup_dir, f'{sid}_backup.dmp'), 'full=y'], check=True)
    except subprocess.CalledProcessError as e:
        logging.error(f"Error occurred during Oracle restore: {e}")

def restore_mongodb(host, port, username, password, database, backup_dir):
    try:
        # 使用安全的参数传递方式
        subprocess.run(['mongorestore', '--host', host, '--port', str(port), '--username', username, '--password', password, '--db', database, backup_dir], check=True)
    except subprocess.CalledProcessError as e:
        logging.error(f"Error occurred during MongoDB restore: {e}")

File: DB/db_backup_restore/go_bak_restore/restore.go
package main

import (
	"fmt"
	"os/exec"
)

// restoreMySQL 函数还原 MySQL 数据库
func restoreMySQL(config DatabaseConfig, backupFile string) error {
	cmd := exec.Command("mysql", "-h", config.Host, "-u", config.Username, "-p"+config.Password, config.Database, "<", backupFile)

	err := cmd.Run()
	if err != nil {
		return fmt.Errorf("error occurred during MySQL restore: %s", err)
	}
	return nil
}

// restoreOracle 函数还原 Oracle 数据库
func restoreOracle(config DatabaseConfig, backupFile string) error {
	cmd := exec.Command("imp", config.Username+"/"+config.Password+"@"+config.SID, "file="+backupFile, "full=y")

	err := cmd.Run()
	if err != nil {
		return fmt.Errorf("error occurred during Oracle restore: %s", err)
	}
	return nil
}

// restoreMongoDB 函数还原 MongoDB 数据库
func restoreMongoDB(config DatabaseConfig, backupDir string) error {
	cmd := exec.Command("mongorestore", "--host", config.Host, "--port", strconv.Itoa(config.Port), "--username", config.Username, "--password", config.Password, "--db", config.Database, backupDir)

	err := cmd.Run()
	if err != nil {
		return fmt.Errorf("error occurred during MongoDB restore: %s", err)
	}
	return nil
}
